mobile/v1/bigquery/sql-runner/sql/standard/03-optional-modules/01-app-errors/01-main/01-app-errors.sql

```diff
@@ -0,0 +1,111 @@
+/*
+   Copyright 2021 Snowplow Analytics Ltd. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+{{if eq (or .enabled false) true}}
+  -- App errors events schema evolved over time. Finding all versions of the column
+  DECLARE APP_ERRORS_EVENTS_COLUMNS, APP_ERRORS_QUERY STRING;
+  CALL {{.output_schema}}.mobile_app_errors_fields(APP_ERRORS_EVENTS_COLUMNS);
+
+  SET APP_ERRORS_QUERY = format("""
+    CREATE OR REPLACE TABLE {{.scratch_schema}}.mobile_app_errors_this_run{{.entropy}}
+    PARTITION BY DATE(derived_tstamp)
+    AS (
+
+      SELECT
+        e.event_id,
+
+        e.app_id,
+
+        e.user_id,
+        e.device_user_id,
+        e.network_userid,
+
+        e.session_id,
+        e.session_index,
+        e.previous_session_id,
+        e.session_first_event_id,
+
+        e.dvce_created_tstamp,
+        e.collector_tstamp,
+        e.derived_tstamp,
+
+        --Error details
+        %s,
```

Comment:
Feels more future-proof to have this at the end of the query, since that allows us to handle additional fields in future without a table-definition headache.

Reply:
Makes sense to mitigate future pain. Deviates from the RS model then not the end of the world. Will change